basketserver/src/main/java/de/upb/crypto/incentive/basketserver/BasketController.java

```diff
@@ -0,0 +1,155 @@
+package de.upb.crypto.incentive.basketserver;
+
+import de.upb.crypto.incentive.basketserver.exceptions.*;
+import de.upb.crypto.incentive.basketserver.model.Basket;
+import de.upb.crypto.incentive.basketserver.model.Item;
+import de.upb.crypto.incentive.basketserver.model.requests.PayBasketRequest;
+import de.upb.crypto.incentive.basketserver.model.requests.PutItemRequest;
+import de.upb.crypto.incentive.basketserver.model.requests.RedeemBasketRequest;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.UUID;
+
+
+/**
+ * A REST controller that defines and handles all requests of the basket server.
+ */
+@AllArgsConstructor
+@RestController
+public class BasketController {
+
+    private BasketService basketService;  // Spring boot automatically injects a BasketService object
+
+    /**
+     * Can be used for health checking.
+     */
+    @GetMapping("/")
+    String getHelloWorld() {
+        return "Hello World";
+    }
+
+
+    /**
+     * Query all shopping items that can be purchased
+     */
+    @GetMapping("/items")
+    Item[] getAllBasketItems() {
+        return basketService.getItems();
+    }
+
+    /**
+     * Create a new basket
+     */
+    @GetMapping("/basket/new")
+    UUID initializeBasket() {
+        return basketService.createNewBasket();
+    }
+
+    /**
+     * Query a basket by its id
+     */
+    @GetMapping("/basket")
+    ResponseEntity<Basket> getBasket(@RequestHeader UUID basketId) throws BasketServiceException {
+        var basket = basketService.getBasketById(basketId);
```

Comment:
Clarification: Why do you like `var` so much?

Reply:
Less code to write with the same level of type safety. And I like the fact that several var statements have aligned variable names, but this is just a personal preference.
If you prefer typing the actual type I can of course change this.

Reply:
Thanks for the insight. Please do not change.